atomix/cluster/src/test/java/io/atomix/raft/storage/log/RaftLogTest.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright © 2020 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.raft.storage.log;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.atomix.cluster.MemberId;
+import io.atomix.raft.cluster.RaftMember.Type;
+import io.atomix.raft.cluster.impl.DefaultRaftMember;
+import io.atomix.raft.storage.log.entry.ApplicationEntry;
+import io.atomix.raft.storage.log.entry.ConfigurationEntry;
+import io.atomix.raft.storage.log.entry.InitialEntry;
+import io.atomix.raft.storage.log.entry.RaftLogEntry;
+import java.io.File;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.time.Instant;
+import java.util.Set;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+class RaftLogTest {
+  @TempDir File directory;
+
+  private final InitialEntry initialEntry = new InitialEntry();
+  private final ConfigurationEntry configurationEntry =
+      new ConfigurationEntry(
+          1234L,
+          Set.of(
+              new DefaultRaftMember(MemberId.from("0"), Type.ACTIVE, Instant.ofEpochSecond(1234))));
+  private final ByteBuffer data = ByteBuffer.allocate(8);
+  private final ApplicationEntry applicationEntry = new ApplicationEntry(1, 2, data);
+  private RaftLog raftlog;
+  private RaftLogReader reader;
+
+  @BeforeEach
+  void setup() {
+    raftlog = RaftLog.builder().withDirectory(directory).withName("test").build();
+    reader = raftlog.openReader();
+    data.order(ByteOrder.LITTLE_ENDIAN).putInt(123456);
+  }
+
+  @AfterEach
+  void tearDown() {
+    reader.close();
+    raftlog.close();
+  }
+
+  @Test
+  void shouldAppendInitialEntry() {
+    // given
+    final RaftLogEntry entry = new RaftLogEntry(1, initialEntry);
+    // when
+    final var appended = raftlog.append(entry);
+
+    // then
+    assertThat(reader.hasNext()).isTrue();
+    final var entryRead = reader.next();
+    assertThat(entryRead.term()).isEqualTo(1);
+    assertThat(entryRead.entry()).isInstanceOf(InitialEntry.class);
+    assertThat(appended).isEqualTo(entryRead);
+  }
+
+  @Test
+  void shouldAppendConfigurationEntry() {
+    // given
+    final RaftLogEntry entry = new RaftLogEntry(1, configurationEntry);
+    // when
+    final var appended = raftlog.append(entry);
```

Comment:
```suggestion
    raftlog.append(entry);
```

The variable is unused